src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Figure['shape'] = 'triangle';
 
-}
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].some((side) => side <= 0)) {
+      throw new Error('A side can\'t have length <= 0');
+    }
+
+    if (a >= (b + c) || b >= (a + c) || c >= (a + b)
+    ) {
+      throw new Error(
+        'The longest side of a triangle is >= than a sum of two others',
+      );
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const { a, b, c } = this;
 
+    const s = (a + b + c) / 2;
+    const area = Math.sqrt(s * (s - a) * (s - b) * (s - c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Figure['shape'] = 'circle';
+
+  constructor(
+    public color: Figure['color'],
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('A radius can\'t be <= 0');
+    }
+  }
 
+  getArea(): number {
+    const { radius } = this;
+    const area = Math.PI * (radius ** 2);
+
+    return Math.floor(area * 100) / 100;
```

Comment:
'Math.floor(area * 100) / 100' is duplicated, so make sense to move it to the separate function